pkg/gatherers/clusterconfig/gather_cluster_version_test.go

```diff
@@ -0,0 +1,134 @@
+package clusterconfig
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/openshift/insights-operator/pkg/utils/marshal"
+
+	configFake "github.com/openshift/client-go/config/clientset/versioned/fake"
+	"k8s.io/client-go/kubernetes/fake"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_getClusterVersion(t *testing.T) {
+	lastTimestampEvent := metav1.Time{Time: time.Now().Add(2)}
+	compactEvents := eventListToCompactedEventList(&v1.EventList{
+		Items: []v1.Event{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "openshift-cluster-version",
+				},
+				Reason:        "Unit test",
+				Message:       "This is a unit test event",
+				LastTimestamp: lastTimestampEvent,
+				Count:         1,
+			},
+		},
+	})
+
+	tests := []struct {
+		name                     string
+		clusterVersionDefinition *configv1.ClusterVersion
+		pods                     *v1.PodList
+		events                   *v1.EventList
+		wantRecords              []record.Record
+		wantErrCounts            int
+		interval                 time.Duration
+	}{
+		{
+			name: "successful retrieve node version",
+			clusterVersionDefinition: &configv1.ClusterVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "version",
+				},
+				Spec: configv1.ClusterVersionSpec{
+					ClusterID: "cluster-id",
+					Channel:   "stable-4.13",
+				},
+			},
+			pods: &v1.PodList{
+				Items: []v1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "version",
+							Namespace: "openshift-cluster-version",
+						},
+						Status: v1.PodStatus{
+							InitContainerStatuses: []v1.ContainerStatus{{RestartCount: 1}},
+						},
+					},
+				},
+			},
+			events: &v1.EventList{
+				Items: []v1.Event{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: "openshift-cluster-version",
+						},
+						Reason:        "Unit test",
+						Message:       "This is a unit test event",
+						LastTimestamp: lastTimestampEvent,
+						Count:         1,
+					},
```

Comment:
Just detail. This event definition repeats here so you could extract it to some variable as well and save few lines of code.

Reply:
Done. I do have a feeling that this isn't something that we should do... But, I have no logical argument to say "why"... and in this specific case, I don't see benefits to not doing it. In my mind, it feels like the data in table-driven tests should test specific and the dups are expected and acceptable. But, as I said... I have no logical argument, just this feeling.